### ISSUE TITLE
Unable to use an array for data other than a constant.

### ISSUE THREAD

**Issue by pedroknup** [mbrn/material-table#259]
I created my MaterialTable with some static data, everything was great. But when I started to fetch data from api, I'm getting this error:

![image](https://user-images.githubusercontent.com/15652671/53349322-9fc5d500-38fb-11e9-8571-b3efbf100347.png)

Using the same function on example to create an array

`const rows = [
    createData('14:25 21/02', "Delivery", 0, 'Pedro Knup', '(31) 98707-1755', 'Av Falcão 625', 'R$12,00'),
    createData('14:13 21/02', "Delivery", 2, 'Pedro Knup', '(31) 98707-1755', 'Rua Tiradentes 80A', 'R$22,00'),
    createData('14:11 21/02', "Delivery", 3, 'Pedro Knup', '(31) 98707-1755', 'Av Paulista 100', 'R$42,00'),
    createData('19:50 20/02', "Balcão", 1, 'Pedro Knup', '(31) 98707-1755', 'Av Falcão 625', 'R$15,00'),
];`

and on the table definition:
`data={rows}`

everything works

but when I set 
`data={this.state.pedidos}`
with empty array as initial value and then I set this 'pedidos' state as an filled array, I got that error above.

The funny thing is that if I set this state as the same array that was working before (const rows) I got the same error.

So for now I can't use this component since I can't use a dynamic array for its data props. 

Any tips?

**Comment by mbrn** [comment]
Hi @pedroknup ,

Can you try this?

call your api in componentDidMount() method and set your data to state. And use it in material-table. 

If it is not working still, you can share code.

**Comment by pedroknup** [comment]
> Hi @pedroknup ,
> 
> Can you try this?
> 
> call your api in componentDidMount() method and set your data to state. And use it in material-table.
> 
> If it is not working still, you can share code.

I'm calling my API and setting the state on componentDidMount() already.

My jsx file looks like this

````
let id = 0;
function createData(data, entrega, status, cliente, celular, endereco, total) {
    id += 1;
    return { data, pedido: id, entrega, status, cliente, celular, endereco, total };
}

const rows = [
    createData('14:25 21/02', "Delivery", 0, 'Pedro Knup', '(31) 98707-1755', 'Av Falcão 625', 'R$12,00'),
    createData('14:13 21/02', "Delivery", 2, 'Pedro Knup', '(31) 98707-1755', 'Rua Tiradentes 90A', 'R$22,00'),
    createData('14:11 21/02', "Delivery", 3, 'Pedro Knup', '(31) 98707-1755', 'Av Paulista 100', 'R$42,00'),
    createData('19:50 20/02', "Balcão", 1, 'Pedro Knup', '(31) 98707-1755', 'Av Falcão 625', 'R$15,00'),
];

class Pedidos extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            pedidos:[],
            isLoading:false
        }
        this.fetchData = this.fetchData.bind(this);
    }
    fetchData() {
         var pedidos = [];
         //calling my api and filling the array
         this.setState({pedidos});
     }
     render() {
        return (
...
                    <MaterialTable
                            isLoading={this.state.isLoading}
                            options={
                                {
                                    toolbar: false,
                                }
                            }
                            localization={tableTranslation
                            }
                            columns={[
                                { title: 'Pedido', field: 'pedido' },
                                {
                                    title: 'Entrega', field: 'entrega',
                                },
                                { title: 'Cliente', field: 'cliente' },
                                { title: 'Celular', field: 'celular' },
                                { title: 'Endereço', field: 'endereco' },
                                { title: 'Total', field: 'total' },

                            ]}

                            data={this.state.pedidos}
                     />

...

export default Pedidos();

**Comment by mbrn** [comment]
@pedroknup ,

Look at this. https://stackblitz.com/edit/react-rpbcta

**Comment by pedroknup** [comment]
> @pedroknup ,
> 
> Look at this. https://stackblitz.com/edit/react-rpbcta

It worked, @mbrn!

First I deleted all unrelated things and I noticed that a dialog (with no relation with 'pedidos') was causing this error. 

Then I could render with your fetchData() method you sent me. But again, when I changed to my API response array, I got the same error.

But then I noticed that I was passing an object into each array index, which was causing this error (although before I came here I tried to remove them, letting just the string values).

Thank you for your help.

:v:
